src/ase/cw/model/Server.java

```diff
@@ -152,7 +176,18 @@ public void run() {
                 }
                 //Order finished
                 orderHandler.orderFinished(currentOrder, Server.this);
+
+                //Pause processing if server on break
+                synchronized (serverThread) {
+                    if (pauseThread) {
+                        try {
+                            serverThread.wait();
```

Comment:
If the server is paused, is it still busy?
One suggestion:use a enum instead of with three states: WORKING, PAUSE, FREE.
and instead of isBusy you could use getState.

Reply:
You could also add a method in OrderConsumer: onServerStateChanged notify the listener when the server change the state